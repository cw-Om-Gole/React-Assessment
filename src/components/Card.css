
.card {
    width: 100%;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    border-radius: 10px;
}

.card .card-image {
    width: 100%;
    height: auto;
    object-fit: cover;
    border-radius: 10px 10px 0px 0px;
}

.card-content {
    padding: 16px;
}

.category {
    font-size: 12px;
    letter-spacing: 1px;
    color: #9CA3AF;
    margin-bottom: 8px;
}

.card-title {
    font-size: 18px;
    font-weight: 600;
    color: #111827;
    margin-bottom: 12px;
    margin-top: 0px;
}

.card-description {
    font-size: 14px;
    color: #4B5563;
    margin-bottom: 16px;
}

.card-footer {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
}

.learn-more {
    color: #6366F1;
    display: inline-flex;
    align-items: center;
    text-decoration: none;
    font-size: 14px;
    cursor: pointer;
    margin-bottom: 8px;
}

.learn-more:hover {
    text-decoration: underline;
}

.arrow-icon {
    width: 16px;
    height: 16px;
    margin-left: 8px;
}

.views, .comments {
    color: #9CA3AF;
    display: inline-flex;
    align-items: center;
    font-size: 14px;
    margin-right: 12px;
    padding-right: 12px;
    border-right: 2px solid #E5E7EB;
}

.comments {
    border-right: none;
}

.view-icon, .comment-icon {
    width: 16px;
    height: 16px;
    margin-right: 4px;
}

.btn {
    background-color: rgb(239, 42, 42);
    color: white;
    display: block;
    width: 100%;
    text-align: center;
    border-radius: 10px;
    padding: 10px;
}

.btn:hover {
    cursor: pointer;
    background-color: rgb(187, 16, 16);

}