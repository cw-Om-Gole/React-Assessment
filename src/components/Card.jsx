import React from 'react';
import './Card.css';

const Card = ({carDetails}) => {


    const getCarLocation = (carDetails) => {
        return carDetails.areaName + "," + " " + carDetails.cityName;
    }

    return (
        <>
                <div className="card">
                    <img className="card-image" src={carDetails.imageUrl ? carDetails.imageUrl : "https://dummyimage.com/300x225/000/fff"} alt="blog"/>
                        <div className="card-content">
                            <h2 className="card-title">{carDetails.carName}</h2>
                            <p className="card-description">
                                {carDetails.km} km
                                &nbsp;&nbsp;|&nbsp;&nbsp;
                                {carDetails.fuel}
                                &nbsp;&nbsp;|&nbsp;&nbsp;
                                {getCarLocation(carDetails)}
                            </p>
                            <h2 className="card-price">{carDetails.price}</h2>
                            <div className="card-footer">
                                <a className="btn">Get Seller Details</a>
                            </div>
                        </div>
                </div>
        </>
    )
}

export default Card;
