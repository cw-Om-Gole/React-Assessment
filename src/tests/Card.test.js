import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; 
import Card from '../components/Card';

describe('Card Component Snapshot', () => {
    const mockCarDetails = {
        carName: 'Toyota Prius',
        km: '12000',
        fuel: 'Petrol',
        areaName: 'Vashi',
        cityName: 'Mumbai',
        price: 25000,
        imageUrl: 'https://dummyimage.com/300x225/000/fff',
    };

    it('matches the snapshot', () => {
        const cardComponent = render(<Card carDetails={mockCarDetails} />);
        expect(cardComponent).toMatchSnapshot();
    });

    it('renders the card with car details', () => {
        render(<Card carDetails={mockCarDetails} />);

        const image = screen.getByRole('img');
        expect(image).toHaveAttribute('src', mockCarDetails.imageUrl);

        const carName = screen.getByText('Toyota Prius');
        expect(carName).toBeInTheDocument();

        const kilometers = screen.getByText(/12000 km/);
        expect(kilometers).toBeInTheDocument();

        const fuelType = screen.getByText(/Petrol/);
        expect(fuelType).toBeInTheDocument();

        const location = screen.getByText(/Vashi, Mumbai/);
        expect(location).toBeInTheDocument();


        const price = screen.getByText(/25000/);
        expect(price).toBeInTheDocument();

        const button = screen.getByText('Get Seller Details');
        expect(button).toBeInTheDocument();
    });
});