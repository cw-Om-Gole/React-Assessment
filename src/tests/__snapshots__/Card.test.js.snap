// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Snapshot matches the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <img
          alt="blog"
          class="card-image"
          src="https://dummyimage.com/300x225/000/fff"
        />
        <div
          class="card-content"
        >
          <h2
            class="card-title"
          >
            Toyota Prius
          </h2>
          <p
            class="card-description"
          >
            12000
             km   |  
            Petrol
              |  
            Vashi, Mumbai
          </p>
          <h2
            class="card-price"
          >
            25000
          </h2>
          <div
            class="card-footer"
          >
            <a
              class="btn"
            >
              Get Seller Details
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="blog"
        class="card-image"
        src="https://dummyimage.com/300x225/000/fff"
      />
      <div
        class="card-content"
      >
        <h2
          class="card-title"
        >
          Toyota Prius
        </h2>
        <p
          class="card-description"
        >
          12000
           km   |  
          Petrol
            |  
          Vashi, Mumbai
        </p>
        <h2
          class="card-price"
        >
          25000
        </h2>
        <div
          class="card-footer"
        >
          <a
            class="btn"
          >
            Get Seller Details
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
